app/assets/javascripts/beak/widgets/ractives/widget.coffee

```diff
@@ -190,32 +206,79 @@ RactiveWidget = RactiveDraggableAndContextable.extend({
                 event
             realEvent.run(this, widget)
 
+          notifyEventName = if isNewWidget then 'new-widget-finalized' else 'widget-updated'
+          @fire(notifyEventName, widget.id, widget.type, @getExtraNotificationArgs()...)
+
           @fire('update-widgets')
 
       catch ex
         console.error(ex)
       finally
         return false
 
+    "*.stop-widget-drag": (_) ->
+      @notifyWidgetMoved()
+
   }
 
   # coffeelint: disable=max_line_length
   partials: {
-    editorOverlay: """
-                   {{ #isEditing }}
-                     <div draggable="true" style="{{dims}}"
-                          class="editor-overlay{{#isSelected}} selected{{/}}{{#widget.type === 'plot'}} plot-overlay{{/}}"
-                          on-click="@this.fire('hide-context-menu') && @this.fire('select-widget', @event)"
-                          on-contextmenu="@this.fire('show-context-menu', @event)"
-                          on-dblclick="@this.fire('edit-widget')"
-                          on-dragstart="start-widget-drag"
-                          on-drag="drag-widget"
-                          on-dragend="stop-widget-drag"></div>
-                   {{/}}
-                   """
+    editorOverlay:
+      """
+      {{ #isEditing }}
+      <div
+        draggable="true"
+        style="{{dims}}"
+        class="editor-overlay{{#isSelected}} selected{{/}}{{#widget.type === 'plot'}} plot-overlay{{/}}"
+        on-click="@this.fire('hide-context-menu') && @this.fire('select-widget', @event)"
+        on-contextmenu="@this.fire('show-context-menu', @event)"
+        on-dblclick="@this.fire('edit-widget')"
+        on-dragstart="start-widget-drag"
+        on-drag="drag-widget"
+        on-dragend="stop-widget-drag">
+      </div>
+      {{/}}
+      """
+  }
+
+})
+
+RactiveValueWidget = RactiveWidget.extend({
```

Comment:
I don't know... it just feels like this should simply be in a separate file, and not be something that needs to be yanked out of the export object for `RactiveWidget`.

Reply:
Sure.